libtenzir/builtins/operators/python.cpp

```diff
@@ -445,13 +445,15 @@ class python_operator final : public crtp_operator<python_operator> {
 
 private:
   config config_ = {};
+  mutable bp::environment env_ = {};
```

Comment:
This will always be default-initialized in the operator when you actually run it. As part of the optimization process, the operator goes through an inspect roundtrip, and this just doesn't make it past that.

The call operator of `crtp_operator` is also marked `const` for a very good reason. If the state must be mutable then it should be on the coroutine stack, not outside of it.This will always be default-initialized in the operator when you actually run it. As part of the optimization process, the operator goes through an inspect roundtrip, and this just doesn't make it past that.

The call operator of `crtp_operator` is also marked `const` for a very good reason. If the state must be mutable then it should be kept on the coroutine stack, not outside of it.

Reply:
I forgot about the inspect roundtrip. Fixing that would mean writing an overload for `boost::process::environment`. I opted to implement a slightly more expensive but much simpler approach now.